LibSWBF2.NET/Utils/MarshalingUtils.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Runtime.InteropServices.WindowsRuntime;
+using LibSWBF2.Wrappers;
+
+namespace LibSWBF2.Utils
+{
+    class MemUtils {
+
+
+        public static T[] IntPtrsToWrappers<T>(IntPtr nativePtr, int count) where T : NativeWrapper, new()
+        {
+            if (nativePtr == IntPtr.Zero) return new T[0];
+
+            T[] wrappers = new T[count];
+            IntPtr[] ptrArr = new IntPtr[count];
+            Marshal.Copy(nativePtr, ptrArr, 0, count);
+            
+            for (int i = 0; i < count; i++){
+                wrappers[i] = new T();
+                wrappers[i].SetPtr(ptrArr[i]);
+            }
+
+            return wrappers;
+        }
+        
+
```

Comment:
I feel like there's a way to do this without generics?  Then again C# seems more restricted than what I'm used to w/C++ generics (ie, ```new T()``` isn't doable by default in C#)